operator/src/main/java/org/bf2/operator/managers/CapacityManager.java

```diff
@@ -0,0 +1,330 @@
+package org.bf2.operator.managers;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import lombok.Getter;
+import lombok.Setter;
+import org.bf2.common.ConditionUtils;
+import org.bf2.common.ManagedKafkaResourceClient;
+import org.bf2.common.OperandUtils;
+import org.bf2.operator.operands.AbstractKafkaCluster;
+import org.bf2.operator.operands.KafkaInstanceConfigurations;
+import org.bf2.operator.operands.KafkaInstanceConfigurations.InstanceType;
+import org.bf2.operator.operands.OperandReadiness;
+import org.bf2.operator.resources.v1alpha1.ManagedKafka;
+import org.bf2.operator.resources.v1alpha1.ManagedKafkaAgent;
+import org.bf2.operator.resources.v1alpha1.ManagedKafkaCondition;
+import org.bf2.operator.resources.v1alpha1.ManagedKafkaCondition.Reason;
+import org.bf2.operator.resources.v1alpha1.ManagedKafkaCondition.Status;
+import org.bf2.operator.resources.v1alpha1.Profile;
+import org.bf2.operator.resources.v1alpha1.ProfileCapacity;
+import org.bf2.operator.resources.v1alpha1.ProfileCapacityBuilder;
+import org.jboss.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import java.net.HttpURLConnection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Responsible for ensuring the resources for a given ManagedKafka are available
+ * prior to allowing the deployment to proceed.
+ */
+@ApplicationScoped
+public class CapacityManager {
```

Comment:
@shawkins I think we've to exclude reserved mk from remainingUnits calculations.

At the moment if you've a max nodes N, then `maxUnits = N/unitPerNode` , where `unitPerNode` is either `0.1` for developer instance or `3` for standard instance. This is okay. 

And the remainingUnits is calculate against `maxUnits - usedMKCapacityCount`. The `usedMKCapacityCount` at the moment also includes the reserved MK while it should only include real deployment. That's to say that, when we are deploying a reserved instance, we should claim resources. And when we are deleting a reserved instance we shouldn't release resources.
If we do not do so, we end up in a situation where for example with 3 nodes for standard instance. If you prewarm them initially with 1 reserved instance, then remainingUnits becomes `0`, and any subsequent `real` deployment gets rejected due to insufficient resource